docs/literate/tutorial/two_stage_analysis.jl

```diff
@@ -0,0 +1,220 @@
+# # Two-stage EEG analysis using Unfold & UnfoldStats
+
+# ## 0. Setup
+# Load required packages. 
+using UnfoldStats
+using Unfold
+using DataFrames
+using Chain
+using Statistics
+using HypothesisTests
+using CairoMakie
+using UnfoldMakie
+
+# ## 1. Simulate data
+# This section can be skipped, if one already has (real) data that they want to analyse.
+
+# ```@raw html
+# <details>
+# <summary>Click to expand the simulation details</summary>
+# ```
+
+using UnfoldSim
+using StableRNGs
+
+design = MultiSubjectDesign(
+    n_subjects = 20,
+    n_items = 100,
+    items_between = Dict(:continuous => range(0, 2, length = 10)),
+)
+
+β = [1, 1, 0.5, 0.2]
+σs = Dict(:subject => [1, 0.1, 0.1, 0.1])
+
+signal = MixedModelComponent(;
+    basis = UnfoldSim.hanning(50),
+    formula = @formula(
+        0 ~
+            1 +
+            continuous +
+            continuous^2 +
+            continuous^3 +
+            (1 + continuous + continuous^2 + continuous^3 | subject)
+    ),
+    β = β,
+    σs = σs,
+)
+
+hart = headmodel(type = "hartmut")
+signal_multichannel = MultichannelComponent(signal, hart => "Left Postcentral Gyrus")
+
+onset = UniformOnset(; width = 50, offset = 60)
+noise = PinkNoise(; noiselevel = 5)
+
+# ```@raw html
+# </details >
+# ```
+
+data, events = simulate(
+    StableRNG(1),
+    design,
+    signal_multichannel,
+    onset,
+    noise,
+    return_epoched = false,
+);
+
+# ```@raw html
+# <details>
+# <summary>Click to expand event data frame </summary>
+# ```
+first(events, 12)
+# ```@raw html
+# </details >
+# ```
+
+# ## 2. Fit an Unfold model for each subject (first stage)
+# In the first stage, we fit an Unfold model for each subject separately.
+
+## Specify a temporal basis function
+basisfunction = firbasis((-0.1, 0.7), 100)
+
+## Specify the model formula
+formula = @formula 0 ~ 1 + spl(continuous, 4)
+
+## Combine basisfunction and formula in an event dict
+event_dict = Dict(Any => (formula, basisfunction))
+
+subject_list = unique(events.subject)
+model_list = UnfoldLinearModelContinuousTime[]
+
+## Slice the data by its last dimension (i.e. the subject dimension)
+data_slices = eachslice(data, dims = ndims(data))
+
+for s = 1:size(data, ndims(data))
+    m = fit(
```

Comment:
I wonder if something like this would also work

```
fit.(UnfoldModel,event_dict,groupby(events,:subject),eachslice(data,dims=ndims(data));show_progress=false)
```
(outside of review, your version is fine and explicit)

Reply:
I tried it (with some minor adaptions) but I get the following `ArgumentError`:
```julia-repl
julia> fit.(UnfoldModel,Ref(event_dict),groupby(events,:subject),data_slices)
ERROR: ArgumentError: broadcasting over `GroupedDataFrame`s is reserved
Stacktrace:
 [1] broadcastable(::GroupedDataFrame{DataFrame})
   @ DataFrames ~/.julia/packages/DataFrames/58MUJ/src/groupeddataframe/groupeddataframe.jl:326
 [2] map
   @ Base ./tuple.jl:292 [inlined]
 [3] broadcasted(::typeof(fit), ::Type, ::Base.RefValue{…}, ::GroupedDataFrame{…}, ::Slices{…})
   @ Base.Broadcast ./broadcast.jl:1346
 [4] top-level scope
   @ REPL[718]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

Would be nice if it worked. Maybe it's possible when we adapt the syntax a bit.

Reply:
One solution is to use `collect` on the grouped data frame.
`fit.(UnfoldModel,Ref(event_dict), collect(DataFrame, groupby(events,:subject)),data_slices)`

Reply:
nice! that looks pretty sleek. Thanks for checking - maybe something for @ReneSkukies  to keep in mind for UnfoldBIDS?

Reply:
I did a small comparison using `BenchmarkTools`. I would say the performance is comparable.

The functions:
```julia
function fit_Unfold_models1(data, events, event_dict)

	model_list = UnfoldLinearModelContinuousTime[]
	data_slices = eachslice(data, dims = ndims(data))
	
	for s = 1:size(data, ndims(data))
	    m = fit(
	        UnfoldModel,
	        event_dict,
	        subset(events, :subject => ByRow(==(subject_list[s]))),
	        data_slices[s],
	    )
	    push!(model_list, m)
	end
	return model_list
end
```

```julia
function fit_Unfold_models2(data, events, event_dict)
	data_slices = eachslice(data, dims = ndims(data))
	model_list = fit.(UnfoldModel,Ref(event_dict), collect(DataFrame, groupby(events,:subject)),data_slices)

	return model_list
end
```

The benchmark:
![grafik](https://github.com/unfoldtoolbox/UnfoldStats.jl/assets/22366977/9d854ac1-3e0d-4a11-b8a4-6503a1d907d9)